### PR TITLE
fraud: update rsmt2d to change isRow->Axis

### LINKED ISSUE #669: fraud: Change `isRow` check to const value from `rsmt2d` library
We are using isRow flag to detect where verification was failed. In the newest version of rsmt2d bool flag was reworked on enum. So we need to adapt our logic to these changes.

After updating to a newer version of rsmt2d we should change `isRow` to corresponding [Axis](https://github.com/celestiaorg/rsmt2d/blob/master/extendeddatacrossword.go#L12-L15) value from const in the following places: 
* https://github.com/celestiaorg/celestia-node/blob/main/fraud/bad_encoding.go#L28;
* https://github.com/celestiaorg/celestia-node/blob/main/fraud/pb/proof.proto#L11;
* https://github.com/celestiaorg/celestia-node/blob/main/ipld/read.go#L23;

### PR DESCRIPTION
**Pull request by distractedm1nd**
Closes #669. Would this also be all thats needed for #824, @Wondertan?

Also, not sure if the protobuf message was also meant to be updated. Should I keep or revert?

### CODE REVIEW

File: docs/adr/adr-006-fraud-service.md [hunk 1]
```diff
@@ -93,11 +96,17 @@ message ShareWithProof {
...
+   Row = 0;
+   Col = 1;
+}
+
+message BadEncoding {
+   bytes HeaderHash = 1;
+   uint64 Height   = 2;
+   repeated ipld.pb.Share Shares = 3;
+   uint32 Index    = 4;
+   axis Axis    = 5;
```
Comment (adlerjohn) [inline docs/adr/adr-006-fraud-service.md:109]:
```suggestion
   bytes HeaderHash = 1;
   uint64 Height = 2;
   repeated ipld.pb.Share Shares = 3;
   uint32 Index = 4;
   axis Axis = 5;
```

File: docs/adr/adr-006-fraud-service.md [hunk 2]
```diff
@@ -93,11 +96,17 @@ message ShareWithProof {
...
 }
 
-message BadEnconding {
-   required uint64 Height = 1;
-   repeated Share Shares = 2;
-   uint8 Index = 3;
-   bool isRow = 4;
+enum axis {
+   Row = 0;
+   Col = 1;
```
Comment (adlerjohn) [inline docs/adr/adr-006-fraud-service.md:101]:
```suggestion
   ROW = 0;
   COL = 1;
```

File: fraud/pb/proof.proto [hunk 3]
```diff
@@ -4,10 +4,15 @@ package fraud.pb;
 
 import "ipld/pb/share.proto";
 
+enum axis {
+  Row = 0;
+  Col = 1;
```
Comment (adlerjohn) [inline fraud/pb/proof.proto:9]:
```suggestion
  ROW = 0;
  COL = 1;
```

**Review by renaynay — APPROVED** [review]

**Review by adlerjohn — APPROVED** [review]

**Comment by Wondertan** [comment]
@distractedm1nd, gentle ping, are you going to finish opening PRs?

**Comment by distractedm1nd** [comment]
Yes sorry, have been meaning to respond! Getting back from a mini-vacation this week in a few hours, this is the first thing on my list

**Review by adlerjohn — COMMENTED** [review]
merge conflicts

**Comment by distractedm1nd** [comment]
Gah, merged over github because it only looked like two small conflicts, but tons of things broke, fixing now

File: go.sum [hunk 4]
```diff
@@ -178,6 +178,8 @@ github.com/celestiaorg/nmt v0.10.0 h1:HLfVWvpagHz5+uiE0QSjzv350wLhhnybNmrxq9NHLK
 github.com/celestiaorg/nmt v0.10.0/go.mod h1:3bqzTj8xKj0DgQUpOgZzoxvtNkC3MS/hTbQ6dn8SIa0=
 github.com/celestiaorg/rsmt2d v0.5.0 h1:Wa0uNZUXl8lIMJnSunjoD65ktqBedXZD0z2ZU3xKYYw=
 github.com/celestiaorg/rsmt2d v0.5.0/go.mod h1:EZ+O2KdCq8xI7WFwjATLdhtMdrdClmAs2w7zENDr010=
+github.com/celestiaorg/rsmt2d v0.6.0 h1:32Eq5t7lPNbhftPFFjxwCUeEjWg/yGgeMbshxnGw03c=
+github.com/celestiaorg/rsmt2d v0.6.0/go.mod h1:EZ+O2KdCq8xI7WFwjATLdhtMdrdClmAs2w7zENDr010=
```
Comment (Wondertan) [inline go.sum:182]: It is not the end of the world, but go mod tidy is missing.

**Review by Wondertan — APPROVED** [review]

File: go.sum (hunk 4 above)
Comment (distractedm1nd) [reply go.sum]: Oops, thx!

**Review by vgonkivs — APPROVED** [review]

**Comment by vgonkivs** [comment]
LGTM 🚀 
Thank you @distractedm1nd.

**Pull request merged**